### ISSUE TITLE
Catboost crashes iPython kernel after running 2 successful folds

### ISSUE THREAD

**Issue by Paperone80** [catboost/catboost#433]
Problem: **Catboost crashes iPython kernel after running 2 successful folds.**
catboost version: 0.9.1.1 (self-compiled with CLANG 3.8.0)
Operating System: RHEL 7.5, CUDA 9.0.176
CPU: Intel Dual-XEON // 256GB 
GPU: NVIDIA TITAN X 12GB (NVIDIA-SMI 384.81)

Hi, 
I am running Catboost successfully over 2 folds and no matter what I do Catboost crashes the kernel as soon as it starts the 3rd fold. Always. 

Tried to eliminate memory issues but it is not making a change...
Here the code:
```
import os
os.environ['CUDA_VISIBLE_DEVICES'] = '0'
import catboost.gpu._catboost
from catboost import CatBoost, CatBoostClassifier, CatBoostRegressor
catboost.__version__
```

> '0.9.1.1'

```
catboost_params = {
    'devices': '0',
    'task_type': 'GPU', # or 'CPU'
    'verbose': True,  # alias: verbose_eval
    'metric_period': 200,
    'random_seed': SEED, # alias: random_state
    'thread_count': 2,

    ## Reduce GPU memory usage
    'gpu_ram_part': 0.65, # How much of the GPU RAM to use for training.
    'gpu_cat_features_storage': 'CpuPinnedMemory',
    'pinned_memory_size': 30*1024^3,   # GB ^3
    'max_ctr_complexity': 2, # or 2 to save GPU memory usage
    'nan_mode': 'Min', # others: 'Min', 'Max', 'Forbidden'
    'num_boost_round': 20000, # alias: iterations, n_estimators, num_trees
    'learning_rate': 0.05,  # alias: eta
    'max_depth': 3, # alias: depth
    'l2_leaf_reg': 40,
    'boosting_type': 'Plain', # others: 'Ordered'
    'bootstrap_type': 'Bernoulli',  # does not support 'subsample' fraction, other: 'Poisson', 'Bayesian', 'Bernoulli', 'No'
    #'subsample': 0.7,
    #'class_weights': [1, 11],
    #'scale_pos_weight': 5,  # alias class_weights, only logloss loss example class_weights[0.1, 4] for 0 and 1 respectively
    'eval_metric': 'AUC',
    'objective': 'CrossEntropy',  # alias: loss_function
    'od_type': 'Iter',
    'od_wait': 500,  # works like early_stopping but seems buggy
    'allow_writing_files': False
}
```

> [...]
> KFold: 3/4
> Train Shape: (230630, 795)
> Valid Shape: (76877, 795)
> Metric AUC is not implemented on GPU. Will use CPU for metric computation, this could significantly affect learning time

**Crashes the kernel here...**

@annaveronika: Any ideas?

**Comment by annaveronika** [comment]
The crash happens sometimes in the package with GPU training if there is an exception. We have not jet figured out, why there is a crash instead of exception.
The exception happens if there are features in your test dataset with nan values, when there were no nans in this feature in train dataset.

In this case algorithm doesn't know what to do with nans and can only do something random. Thus it is a sanity check.

The correct way to handle this is to add to your train dataset objects with nan values, so that algorithm knows what to do if there are nans when doing prediction.

We are planning to add a parameter to allow the algorithm to threat nans as -infty if there were no nans in train dataset. But I would rather suggest to add nans to train.

We'll write in this issue when we add this flag.

**Comment by Paperone80** [comment]
@annaveronika, thanks a lot for your response. I understand that adding nans to the training set would solve it. However, when k-folding this is not really practical as every column with nans in valid set also need nans in train set. I am using the same folds with XGBoost and LightGBM as well without problems. Oh well, .fillna(0 or else) will do for now.

**Comment by annaveronika** [comment]
closing as duplicate of #105 
We decided to allow this behavoir, we are working on it right now, it'll be soon on github.
